lib/presentation/widget/shared_space_details/add_shared_space_member/add_shared_space_member_viewmodel.dart

```diff
@@ -125,13 +131,46 @@ class AddSharedSpaceMemberViewModel extends BaseViewModel {
     });
   }
 
-  OnlineThunkAction _getSharedSpaceRolesAction() {
+  OnlineThunkAction _updateSharedSpaceMemberRoleAction(
+      SharedSpaceId sharedSpaceId, AccountId accountId, SharedSpaceRoleName newRole) {
     return OnlineThunkAction((Store<AppState> store) async {
-      final roles = (await _getAllSharedSpaceRolesInteractor.execute())
-          .map((result) => result is SharedSpaceRolesViewState ? result.roles : null)
-          .getOrElse(() => []);
+      final rolesList = store.state.sharedSpaceState.rolesList;
+      if (rolesList.isEmpty) {
+        store.dispatch(UpdateSharedSpaceMembersAction(
+            Left<Failure, Success>(UpdateSharedSpaceMemberFailure(RolesListNotFound()))));
+        return;
```

Comment:
I think it is better to return a specific value than to return blank.

Reply:
hmm, it doesn't return a blank. It will notice the toast to notify the user. I missed the toast part implementation for this...